src/blockdata/script.rs

```diff
@@ -1117,6 +1122,20 @@ mod test {
         script = script.push_opcode(opcodes::all::OP_CHECKSIG); comp.push(0xACu8); assert_eq!(&script[..], &comp[..]);
     }
 
+    #[test]
+    fn script_x_only_key() {
+        let mut comp = vec![];
+        let mut script = Builder::new();
+        assert_eq!(&script[..], &comp[..]);
+        // Notice the "20" which prepends the keystr. That 20 is hexidecimal for "32". The Builder automatically adds the 32 opcode
+        // to our script in order to give a heads up to the script compiler that it should add the next 32 bytes to the stack.
+        // To mimic the Builder's automatic insertion of the 32 opcode, we manually add it to the Vec and slice it out in the from_str.
+        // From: https://github.com/bitcoin-core/btcdeb/blob/e8c2750c4a4702768c52d15640ed03bf744d2601/doc/tapscript-example.md?plain=1#L43
+        let keystr = "209997a497d964fc1a62885b05a51166a65a90df00492c8d7cf61d6accf54803be";
+        let x_only_key = XOnlyPublicKey::from_str(&keystr[2..]).unwrap();
+        script = script.push_x_only_key(&x_only_key); comp.extend(Vec::from_hex(keystr).unwrap().iter().cloned()); assert_eq!(&script[..], &comp[..]);
+    }
```

Comment:
The test should be rewritten this way:

```suggestion
    fn script_x_only_key() {
        // Notice the "20" which prepends the keystr. That 20 is hexidecimal for "32". The Builder automatically adds the 32 opcode
        // to our script in order to give a heads up to the script compiler that it should add the next 32 bytes to the stack.
        // From: https://github.com/bitcoin-core/btcdeb/blob/e8c2750c4a4702768c52d15640ed03bf744d2601/doc/tapscript-example.md?plain=1#L43
        let keystr = "209997a497d964fc1a62885b05a51166a65a90df00492c8d7cf61d6accf54803be";
        let x_only_key = XOnlyPublicKey::from_str(&keystr[2..]).unwrap();
        let script = Builder::new().push_x_only_key(&x_only_key); 
        assert_eq!(&script[..], &Vec::from_hex(keystr).unwrap());
    }
```

Reply:
Thanks for the changes!